specification/07-theory_operations.adoc

```diff
@@ -288,6 +288,67 @@ must check that:
 The TSM establishes a secured SPDM session with the physical device DSM by going
 through the steps described in the Secured SPDM Session section.
 
+.Device Connection - SPDM Session Establishment
+[source,mermaid]
+....
+%%{init: {'theme': 'neutral', 'themeVariables': {'darkMode': true}, "flowchart" : { "curve" : "basis" } } }%%
+sequenceDiagram
+
+autonumber
+
+participant DSM as Device DSM or ROT
+participant VMM as Host Supervisor Domain Manager (VMM)
+participant TSM
+
+VMM ->> TSM: [COVH] - sbi_covh_connect_device(device_id, stream_id)
+
+Note over TSM,DSM: TSM: SPDM Connection Setup
+
+TSM ->> TSM: Generate SPDM request: GET_VERSION
+TSM ->> VMM: [COVH] - spdm_req(GET_VERSION)
+VMM ->> DSM: [DOE] - SPDM_GET_VERSION
+DSM ->> VMM: [DOE] - SPDM_VERSION
+VMM ->> TSM: [COVH] - spdm_resp(VERSION)
+
+TSM ->> TSM: Generate SPDM request: GET_CAPABILITIES
+TSM ->> VMM: [COVH] - spdm_req(GET_CAPABILITIES)
+VMM ->> DSM: [DOE] - SPDM_GET_CAPABILITIES
+DSM ->> VMM: [DOE] - CAPABILITIES
+VMM ->> TSM: [COVH] - spdm_resp(CAPABILITIES)
+
+TSM ->> TSM: Generate SPDM request: NEGOTIATE_ALGORITHMS
+TSM ->> VMM: [COVH] - spdm_req(NEGOTIATE_ALGORITHMS)
+VMM ->> DSM: [DOE] - SPDM_NEGOTIATE_ALGORITHMS
+DSM ->> VMM: [DOE] - ALGORITHMS
+VMM ->> TSM: [COVH] - spdm_resp(ALGORITHMS)
+
+TSM ->> TSM: Generate SPDM request: GET_CERTIFICATE
+TSM ->> VMM: [COVH] - spdm_req(GET_CERTIFICATE)
+VMM ->> DSM: [DOE] - GET_CERTIFICATE
+DSM ->> VMM: [DOE] - CERTIFICATE
+VMM ->> TSM: [COVH] - spdm_resp(CERTIFICATE)
+TSM ->> TSM: Verify and store device certificate chain
+
+Note over TSM,DSM: TSM: SPDM Key Exchange
+
+TSM ->> TSM: Generate ephemeral SPDM keys K0
+TSM ->> TSM: Generate SPDM request: KEY_EXCHANGE(K0)
+TSM ->> VMM: [COVH] - spdm_req(KEY_EXCHANGE)
+VMM ->> DSM: [DOE] - SPDM_KEY_EXCHANGE
+DSM ->> VMM: [DOE] - KEY_EXCHANGE_RSP
+VMM ->> TSM: [COVH] - spdm_resp(KEY_EXCHANGE_RSP)
+TSM ->> TSM: Derive DHE secret
+TSM ->> TSM: Generate SPDM request: FINISH
+TSM ->> VMM: [COVH] - spdm_req(FINISH)
```

Comment:
Should we add a step "TSM ->> TSM: Derive SPDM session application secret"

Reply:
For my own curiosity, what are the session application secrets? Just regular secrets (like IDE keys) exchanged over SPDM?

Reply:
Application secret is SPDM session message AEAD keys, which is another set of keys derived from DHE secret indirectly.
In SPDM spec, the message in handshake phase and the message in app phase are using different AEAD keys.

FINISH message is using handshake secret. Once, it is done, handshake phase is over, and session enters application phase.
App secret must be derived after FINISH message, to make sure all application messages are using App secret, not handshake secret.

If we only write "deriver DHE secret, handshake secret", then confusing is "why not deriving application secret".

Reply:
Understood, thanks. Fixed now.